### PR TITLE
Rename `passing/evaluation.*` to `passing/cost_function.*`

### LINKED ISSUE #797: Rename `passing/evaluation.cpp` to `passing/cost_function.cpp`
### Description of the task
The `passing/evaluation.cpp` file is really just a giant cost function used by gradient descent for optimizing passes. As such we should rename it to something like `passing/cost_function.cpp`.

### Acceptance criteria

- [x] Rename `passing/evaluation.cpp` to `passing/cost_function.cpp` and rename all associated tests / usages / comments to match

### PR DESCRIPTION
**Pull request by hardik-ddod**
### Description

Renamed the files related to `ai/passing/evaluation` to `ai/passing/cost_function`, and changed all the related usages.  Also, modified the include statements in `cost_function.h` to accommodate for the change in the path of world module.

### Testing Done

Rebuilt the codebase and ran the pre-existing tests again to make sure any underlying implementations were not affected by the changes in the filenames.

### Resolved Issues

Resolves #797 
### Length Justification

Since the filenames were changed, Git recognizes them as new additions.  However, none of the old code as changed.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by MathewMacDougall** [comment]
@jonathanlew please review

@hardik-ddod Looks like you've got some build errors that will need to be fixed. Check the CI log and you can try build locally.

You'll also need to run a script that auto-formats the code. From the `src/` folder, you can run `./clang_format/fix_formatting.sh -b master`, which will compare the code against your local `master` branch and format anything that changed. This is required for CI.

lmk if you have any questions.

**Review by jonathanlew — CHANGES_REQUESTED** [review]
I know you merged these changes in from before folder structure changes, so please check that you've removed all the files you need to. For example, passing/evaluation.cpp still exists: https://github.com/UBC-Thunderbots/Software/blob/ee577ee366ec82d8c554d398691bdcdc0f914373/src/software/ai/passing/evaluation.cpp

File: src/software/ai/passing/cost_function.cpp [hunk 1]
```diff
@@ -67,6 +67,9 @@ double Passing::ratePass(const World& world, const Passing::Pass& pass,
 double Passing::ratePassShootScore(const Field& field, const Team& enemy_team,
                                    const Passing::Pass& pass)
 {
+    // TODO: You don't even use this first parameter, but stuff is hardcoded below
+    double ideal_shoot_angle_degrees =
+        Util::DynamicParameters::Passing::ideal_min_shoot_angle_degrees.value();
```
Comment (jonathanlew) [inline src/software/ai/passing/cost_function.cpp]: this is a bad merge, please update with the evaluation.cpp on master

File: src/software/ai/passing/cost_function.cpp [hunk 2]
```diff
@@ -294,8 +297,8 @@ double Passing::getStaticPositionQuality(const Field& field, const Point& positi
             static_field_position_quality_friendly_goal_distance_weight.value();
 
     // Make a slightly smaller field, and positive weight values in this reduced field
-    double half_field_length = field.xLength() / 2;
-    double half_field_width  = field.yLength() / 2;
+    double half_field_length = field.length() / 2;
+    double half_field_width  = field.width() / 2;
```
Comment (jonathanlew) [inline src/software/ai/passing/cost_function.cpp]: this is a bad merge, please update with the evaluation.cpp on master

File: src/software/ai/passing/cost_function_test.cpp [hunk 3]
```diff
@@ -98,10 +98,10 @@ TEST_F(PassingEvaluationTest, ratePass_speed_test)
...
              }));
 
-    std::uniform_real_distribution x_distribution(-world.field().xLength() / 2,
-                                                  world.field().xLength() / 2);
-    std::uniform_real_distribution y_distribution(-world.field().yLength() / 2,
-                                                  world.field().yLength() / 2);
+    std::uniform_real_distribution x_distribution(-world.field().length() / 2,
+                                                  world.field().length() / 2);
+    std::uniform_real_distribution y_distribution(-world.field().width() / 2,
+                                                  world.field().width() / 2);
```
Comment (jonathanlew) [inline src/software/ai/passing/cost_function_test.cpp]: this is a bad merge, please update with the evaluation_test.cpp on master

File: src/software/ai/passing/pass_generator.h [hunk 4]
```diff
@@ -36,10 +36,10 @@ namespace Passing
      * == Performance Considerations ==
      * It is important that the pass generator is able to "keep up" with the current
      * time. This is because a pass is defined in part by it's start time. If the
-     * pass generator only completes an interation of pass updates once every 5 seconds,
+     * pass generator only completes an interaction of pass updates once every 5 seconds,
```
Comment (jonathanlew) [inline src/software/ai/passing/pass_generator.h]: please revert

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: src/software/ai/passing/cost_function.cpp (hunk 1 above)
Comment (hardik-ddod) [reply src/software/ai/passing/cost_function.cpp]: I don't think I understand what you mean by that.  Could you elaborate on what you want me to do?

File: src/software/ai/passing/pass_generator.h (hunk 4 above)
Comment (garethellis0) [reply src/software/ai/passing/pass_generator.h]: Sorry, just to clear this up, it's misspelled, but it should be `iteration`, not `interaction`.

**Comment by garethellis0** [comment]
Looks like there was some funky stuff with the merge here. @hardik-ddod, I'll let you and @jonathanlew sort this out before reviewing

**Comment by MathewMacDougall** [comment]
@hardik-ddod any updates on this?

**Comment by MathewMacDougall** [comment]
@hardik-ddod can you resolve these merge conflicts?

**Comment by MathewMacDougall** [comment]
@hardik-ddod any update on this?

**Comment by MathewMacDougall** [comment]
@hardik-ddod Please let us know if you will be updating this or not so we can transfer the work if necessary, since we can't leave things sitting unfinished this long.

**Review by jonathanlew — APPROVED** [review]
Good work fixing the merge conflicts! 💯 🎉 
I assume it's ready for review in this case, but in the future, please ping the reviewers (on slack or tag them in a Github comment) once you've addressed all their comments!

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**